File: src/main/java/com/floorestimatepro/ui/EstimateResultsFrame.java
package com.floorestimatepro.ui;

import java.awt.BorderLayout;
import java.awt.Font;
import java.awt.GridLayout;
import javax.swing.BorderFactory;
import javax.swing.JButton;
import javax.swing.JFileChooser;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JOptionPane;
import javax.swing.JPanel;
import javax.swing.filechooser.FileNameExtensionFilter;

import com.floorestimatepro.export.EstimateExporter;
import com.floorestimatepro.model.EstimateResult;

public class EstimateResultsFrame extends JFrame{
    EstimateResult result;

    public EstimateResultsFrame(EstimateResult result, int roomCount, int obstacleCount){
        this.result = result;
        setTitle("Finished Estimate");
        setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
        setLayout(new BorderLayout(10, 10));

        JPanel panel = new JPanel(new GridLayout(0, 1, 6, 6));
        panel.setBorder(BorderFactory.createEmptyBorder(16, 20, 16, 20));

        JLabel title = new JLabel("Estimate Results");
        title.setFont(title.getFont().deriveFont(Font.BOLD, 18f));
        panel.add(title);
        panel.add(new JLabel("Rooms: " + roomCount));
        panel.add(new JLabel("Obstacles: " + obstacleCount));
        panel.add(new JLabel("Net pixel area: " + round(result.netPixelArea())));
        panel.add(new JLabel("Net area (sq ft): " + round(result.netRealArea())));
        panel.add(new JLabel("Material (sq ft): " + round(result.materialSqFt())));
        panel.add(new JLabel("Estimated cost: $" + round(result.estimatedCost())));

        JPanel buttons = new JPanel();
        JButton exportBtn = new JButton("Export CSV");
        JButton closeBtn = new JButton("Close");
        buttons.add(exportBtn);
        buttons.add(closeBtn);

        exportBtn.addActionListener(e -> onExport());
        closeBtn.addActionListener(e -> dispose());

        add(panel, BorderLayout.CENTER);
        add(buttons, BorderLayout.SOUTH);
        pack();
        setLocationRelativeTo(null);
    }

    void onExport(){
        JFileChooser chooser = new JFileChooser();
        chooser.setFileFilter(new FileNameExtensionFilter("CSV", "csv"));
        if(chooser.showSaveDialog(this) != JFileChooser.APPROVE_OPTION){
            return;
        }
        try{
            String path = chooser.getSelectedFile().getAbsolutePath();
            if(!path.endsWith(".csv")){
                path = path + ".csv";
            }
            EstimateExporter.exportCsv(result, path);
            JOptionPane.showMessageDialog(this, "Exported " + path);
        }catch(Exception ex){
            JOptionPane.showMessageDialog(this, ex.getMessage());
        }
    }

    double round(double value){
        return Math.round(value * 1000.0) / 1000.0;
    }
}

File: python/src/floor_estimate_pro/ui/estimate_results_frame.py
from __future__ import annotations

from PySide6.QtGui import QFont
from PySide6.QtWidgets import (
    QDialog,
    QFileDialog,
    QHBoxLayout,
    QLabel,
    QMessageBox,
    QPushButton,
    QVBoxLayout,
)

from floor_estimate_pro.export.estimate_exporter import EstimateExporter
from floor_estimate_pro.model.estimate_result import EstimateResult


class EstimateResultsFrame(QDialog):
    def __init__(self, result: EstimateResult, room_count: int, obstacle_count: int) -> None:
        super().__init__()
        self.result = result
        self.setWindowTitle("Finished Estimate")

        layout = QVBoxLayout(self)
        title = QLabel("Estimate Results")
        title_font = QFont()
        title_font.setBold(True)
        title_font.setPointSize(18)
        title.setFont(title_font)
        layout.addWidget(title)
        layout.addWidget(QLabel("Rooms: " + str(room_count)))
        layout.addWidget(QLabel("Obstacles: " + str(obstacle_count)))
        layout.addWidget(QLabel("Net pixel area: " + str(self.round(result.net_pixel_area))))
        layout.addWidget(QLabel("Net area (sq ft): " + str(self.round(result.net_real_area))))
        layout.addWidget(QLabel("Material (sq ft): " + str(self.round(result.material_sq_ft))))
        layout.addWidget(QLabel("Estimated cost: $" + str(self.round(result.estimated_cost))))

        buttons = QHBoxLayout()
        export_btn = QPushButton("Export CSV")
        close_btn = QPushButton("Close")
        buttons.addWidget(export_btn)
        buttons.addWidget(close_btn)
        layout.addLayout(buttons)

        export_btn.clicked.connect(self.on_export)
        close_btn.clicked.connect(self.accept)

    def on_export(self) -> None:
        path, _ = QFileDialog.getSaveFileName(self, "Export CSV", "", "CSV (*.csv)")
        if not path:
            return
        try:
            if not path.endswith(".csv"):
                path = path + ".csv"
            EstimateExporter.export_csv(self.result, path)
            QMessageBox.information(self, "Export", "Exported " + path)
        except Exception as ex:
            QMessageBox.warning(self, "Error", str(ex))

    def round(self, value: float) -> float:
        return round(value * 1000.0) / 1000.0
